compiler/src/iree/compiler/Codegen/Common/TileAndDistributeToWorkgroupsPass.cpp

```diff
@@ -255,9 +255,20 @@ struct LowerDispatchWorkgroupCountFromSetEncodingOp
   LogicalResult matchAndRewrite(
       IREE::Flow::DispatchWorkgroupCountFromSetEncodingOp workgroupCountOp,
       PatternRewriter &rewriter) const override {
-    ValueRange workload = workgroupCountOp.getOperands();
+    // The innerTileSizes may not be known at compile time. In that case, for
+    // now we replace any kDynamic value with a dummy reasonable value. Since
+    // this is only for determining the workgroup count, an approximation should
+    // be an acceptable compromise (possibly leaving some performance on the
+    // table) until we figure out how to read runtime-queried tile sizes here.
+    SmallVector<int64_t> innerTileSizeApproximateConstantValues =
```

Comment:
I am not sure we need any of these changes. The inner tile sizes do not factor into the workgroup count calculations. So either the tile sizes have to be set to 0 for iteration dimensions of the op corresponding to the inner tile size (to signify these dimensions arent tiled), or here you should just treat those as untiled/unpartitioned dimensions and set those values to 1.

Looking at the pack op iterator types (https://github.com/iree-org/iree/blob/main/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp#L1810), the loops corresponding to inner tiles dont even feature there. So this is confusing. If this unblocks you in some way then thats fine, but this requires some more investigation.

Reply:
@hanhanW: I have implemented what we discussed yesterday in this new additional commit (added to this existing PR, so please review):
https://github.com/iree-org/iree/pull/11521/commits/8d2c3f99c57df5b82c1c1c54e4e34736fb2251ba

@MaheshRavishankar: thanks for the comments! Like you point out, this code is only interested in the outer dimensions, not in the inner tile dimensions. However, the outer dimensions are themselves the result of a division of the input dimensions by the tile sizes. The existing code calls `PackOp::getResultShape`, which internally performs this division, of the input shape (given by the `workload` values) by the inner tile sizes (where were so far purely static `int64_t` provided by the `materializeEncodingInfo`). As I actually exercise dynamic inner tile sizes in #11522 , these static tile sizes become `-1` and I get a compile error telling me that the evaluation of the `AffineExpr` done internally by `PackOp::getResultShape` is unhappy to find `-1` as a divisor. So I had to provide some other value. In the earlier state of this PR which you commented on, I thought that it would be hard to provide the actual `Value` there so I had settled for a dummy positive static value of `8`, hoping that an inaccurate value would be OK to live with. But yesterday, we discussed this with @hanhanW and found that it wouldn't be hard to pass this pattern the `materializeEncodingValueFn` and the tensor type to call this on, so that it could obtain the proper runtime Values. So now this PR is updated with the above-linked new commit, which implements that.